azurerm_monitor_diagnostic_setting/main.tf

```diff
@@ -8,14 +8,18 @@ terraform {
   backend "azurerm" {}
 }
 
+data "azurerm_subscription" "subscription" {
+}
+
 resource "azurerm_monitor_diagnostic_setting" "monitor_diagnostic_setting" {
   name                       = "${var.name}-diagnostic-setting"
   target_resource_id         = var.target_resource_id
   log_analytics_workspace_id = var.log_analytics_workspace_id
+  storage_account_id         = var.storage_account_id
 
-  eventhub_name                  = var.eventhub_name
-  eventhub_authorization_rule_id = var.eventhub_name == null ? null : var.eventhub_authorization_rule_id
-  storage_account_id             = var.storage_account_id
+  eventhub_name = var.eventhub_name
+  # required due to this issue: https://github.com/terraform-providers/terraform-provider-azurerm/issues/6356
+  eventhub_authorization_rule_id = var.eventhub_authorization_rule != null ? "/subscriptions/${data.azurerm_subscription.subscription.subscription_id}/resourceGroups/${var.eventhub_resource_group_name}/providers/Microsoft.EventHub/namespaces/${var.eventhub_namespace_name}/authorizationrules/${var.eventhub_authorization_rule}" : null
```

Comment:
Why you are not using data to load the rule?

Reply:
Data does not return the correct namespace id: indeed I've also opened an issue in the official azure provider module:
https://github.com/terraform-providers/terraform-provider-azurerm/issues/6356